x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/components/trusted_apps_grid/index.tsx

```diff
@@ -48,50 +54,123 @@ const RootWrapper = styled.div`
 
 export const TrustedAppsGrid = memo(() => {
   const history = useHistory();
+  const dispatch = useDispatch<Dispatch<AppAction>>();
+  const { getAppUrl } = useAppUrl();
+
   const pagination = useTrustedAppsSelector(getListPagination);
   const listItems = useTrustedAppsSelector(getListItems);
   const isLoading = useTrustedAppsSelector(isListLoading);
   const error = useTrustedAppsSelector(getListErrorMessage);
   const location = useTrustedAppsSelector(getCurrentLocation);
-
-  const handleTrustedAppDelete = useTrustedAppsStoreActionCallback((trustedApp) => ({
-    type: 'trustedAppDeletionDialogStarted',
-    payload: { entry: trustedApp },
-  }));
-
-  const handleTrustedAppEdit: TrustedAppCardProps['onEdit'] = useCallback(
-    (trustedApp) => {
-      history.push(
-        getTrustedAppsListPath({
-          ...location,
-          show: 'edit',
-          id: trustedApp.id,
-        })
-      );
-    },
-    [history, location]
-  );
+  const policyListById = useTrustedAppsSelector(mapOfPoliciesById);
 
   const handlePaginationChange: PaginatedContentProps<
     TrustedApp,
-    TrustedAppCardType
+    ArtifactEntryCardType
   >['onChange'] = useTrustedAppsNavigateCallback(({ pageIndex, pageSize }) => ({
     page_index: pageIndex,
     page_size: pageSize,
   }));
 
+  const handleArtifactCardProps = useMemo(() => {
```

Comment:
If this is going to return a function that needs to be memoized, should it be `useCallback` instead of `useMemo`?

Reply:
If I use `useCallback` then I will not be able to keep private state - below, `cachedCardProps` is a private scooped object that the memoized function uses. That was the only reason I used `useMemo()`